app/controllers/concerns/saml_idp_auth_concern.rb

```diff
@@ -64,6 +64,8 @@ def requested_authn_context
     if AppConfig.env.aal_authn_context_enabled == 'true'
       requested_aal_authn_context
     else
+      sp_defined_aal_context = saml_request.requested_aal_authn_context
+      return sp_defined_aal_context if sp_defined_aal_context.present?
       requested_ial_authn_context
```

Comment:
We can simplify this with `.presence` because `a.presence` is the same as `a.present? ? a : nil`:

```suggestion
      saml_request.requested_aal_authn_context || requested_ial_authn_context
```